docs/BOOT_LOADER_SPECIFICATION.md

```diff
@@ -344,57 +392,61 @@ On EFI, any such images shall be added to the list of valid boot entries.
 
 ### Additional Notes
 
-Note that these configurations snippets do not need to be the only
-configuration source for a boot loader. It may extend this list of entries with
-additional items from other configuration files (for example its own native
-configuration files) or automatically detected other entries without explicit
-configuration.
+Note that these boot entry snippets and unified kernels do not need to be the
+only menu entry sources for a boot loader. It may extend this list of
+entries with additional items from other configuration files (for example its
+own native configuration files) or automatically detected other entries without
+explicit configuration.
 
 To make this explicitly clear: this specification is designed with "free"
-operating systems in mind, starting Windows or macOS is out of focus with these
-configuration snippets, use boot-loader specific solutions for that. In the
-text above, if we say "OS" we hence imply "free", i.e. primarily Linux (though
-this could be easily be extended to the BSDs and whatnot).
+operating systems in mind, starting Windows or MacOS is out of focus with these
+boot loader menu entry snippets, use boot-loader specific solutions for
+that. In the text above, if we say "OS" we hence imply "free", i.e. primarily
+Linux (though this could be easily be extended to the BSDs and whatnot).
 
-Note that all paths used in the configuration snippets use a Unix-style "/" as
-path separator. This needs to be converted to an EFI-style "\\" separator in
-EFI boot loaders.
+Note that all paths used in the boot loader menu entry snippets use a
+Unix-style "/" as path separator. This needs to be converted to an EFI-style
+"\\" separator in EFI boot loaders.
 
 
 ## Locating Boot Entries
 
-A _boot loader_ locates `$BOOT` and `$XBOOTLDR`, then simply reads all the
-files `$BOOT/loader/entries/*.conf` and `$XBOOTLDR/loader/entries/*.conf`, and
-populates its boot menu. On EFI, it then extends this with any unified kernel
-images found in `$BOOT/EFI/Linux/*.efi` and `$XBOOTLDR/EFI/Linux/*.efi`. It may
-also add additional entries, for example a "Reboot into firmware" option.
-Optionally it may sort the menu based on the `sort-key`, `machine-id` and
-`version` fields, and possibly others. It uses the file name to identify
-specific items, for example in case it supports storing away default entry
-information somewhere. A boot loader should generally not modify these files.
+A _boot loader_ locates the `$BOOT` partition and the ESP, then simply reads
+all the files `/loader/entries/*.conf` in them, and populates its boot menu
+(and handle gracefully if one of the two partitions is missing, or both are the
+same). On EFI, it then extends this with any unified kernel images found in
```

Comment:
What does it mean "the same"? They have different UUIDs…

Reply:
$BOOT can be the ESP... But OK, here we can change $BOOT to XBOOTLDR, so that they are distinct.